src/mscorlib/src/System/Diagnostics/Eventing/EventSource.cs

```diff
@@ -250,6 +254,24 @@ namespace System.Diagnostics.Tracing
     /// </remarks>
     public partial class EventSource : IDisposable
     {
+
+#if FEATURE_EVENTSOURCE_XPLAT
+        private static XplatEventLogger persistent_Listener = null;
+
+        static EventSource()
```

Comment:
You can double check with @vancem, but adding an explicit cctor like this will force the type to not be beforefieldinit, which will change other perf characteristics. I think it'd be better to have an initialization function and use it at the declaration, like this (sorry for any typos, on my phone):


private readonly static EventListener s_persistentListener = InitializePersistentListener();

private static EventListener InitializePersistentListener()
{
    try 
    {
        if (XplatEventLogger.IsLoggingEnabled())
            return new XplatEventLogger();
    }
    catch {}
    return null;
}